### PR TITLE
Create ModelObjs without swagger definitions.

### PR DESCRIPTION
**Pull request by georgedorn**
Implements creating ModelObjs from autogenerated schema endpoint instead of swagger.

Converts the `models.py` tests to use `vcr` after flailing with `mock` for far too many hours with no luck.

- [x] Improve coverage for the new methods (in particular, the `_get_definitions()` function)
- [x] Add `vcr`-powered tests for spanning relationships and any other behavior the refactor may have broken.
- [x] Fix api_downloads

It turns out that tests of spanning relationships work because they arrange things just so, but those arrangements don't match what the OPTIONS endpoints do now.

The change from swagger also broke `FileFields` - the fix for this involves a change to data_warehouse: https://github.com/pivotbio/data_warehouse/pull/1166

### CODE REVIEW

File: bulk_api_client/model.py [hunk 1]
```diff
@@ -456,16 +456,16 @@ def _get_f(field, properties):
...
-
-
+        getter method
     """
 
     def get_f(cls):
         field_val = cls.data.get(field)
-        if properties[field].get("format") == "uri":
+        if properties[field].get("type") == "foreignkey":
             if "api_download" in field_val:
```
Comment (glassresistor) [inline bulk_api_client/model.py:469]: This isn't going to work but for good reasons.  the type should actually be uri now

File: bulk_api_client/model.py [hunk 2]
```diff
@@ -567,9 +567,12 @@ def with_properties(cls, model_api, uri, data=None):
         class ModelObjWithProperties(cls):
             pass
 
-        model = ".".join([model_api.app.app_label, model_api.model_name])
-        model_properties = model_api.app.client.definitions[model]["properties"]
-        for field, property_dict in model_properties.items():
+        model_properties = ModelObj._get_model_properties(model_api)
+        for field, _ in model_properties.items():
+            if field == "_meta":
```
Comment (glassresistor) [inline bulk_api_client/model.py:572]: I was about to suggest making this slightly more concise by removing it before the list but i like that it visually large

File: bulk_api_client/model.py (hunk 1 above)
Comment (glassresistor) [reply bulk_api_client/model.py:469]: the issue was filetype and fk was overloaded. so you should check this in the field

**Review by glassresistor — CHANGES_REQUESTED** [review]
Great work

File: bulk_api_client/model.py (hunk 1 above)
Comment (georgedorn) [reply bulk_api_client/model.py:469]: Oh, right.  I think maybe I need to also write a test for api_download.

Comment (georgedorn) [reply bulk_api_client/model.py]: That test revealed there wasn't really a safe way (imo) to deal with files without fixing the reported schema on data_warehouse.  Added a PR for that to the description.

**Review by glassresistor — APPROVED** [review]

**Pull request merged**